chain/network/src/peer_manager/peer_store/tests.rs

```diff
@@ -422,3 +439,61 @@ fn test_delete_peers() {
     peer_store.0.lock().delete_peers(&peer_ids);
     assert_peers_in_cache(&peer_store, &[], &[]);
 }
+
+#[test]
+fn test_lru_eviction() {
+    let clock = time::FakeClock::default();
+    let config = make_config_with_cache_size(&[], Default::default(), false, 10);
+    let peer_store = PeerStore::new(&clock.clock(), config).unwrap();
+
+    let (peer_ids, peer_infos): (Vec<_>, Vec<_>) = (0..15)
+        .map(|i| {
+            let id = get_peer_id(format!("node{}", i));
+            let info = get_peer_info(id.clone(), Some(get_addr(i as u16)));
+            (id, info)
+        })
+        .unzip();
+    let peer_addresses = peer_infos.iter().map(|info| info.addr.unwrap()).collect::<Vec<_>>();
+
+    // Fill the peer_store with the first peer_infos
+    peer_store.add_indirect_peers(&clock.clock(), peer_infos[0..10].iter().cloned());
+    assert_peers_in_cache(&peer_store, &peer_ids[0..10], &peer_addresses[0..10]);
+
+    // Push additional peers and check that the most recent peers are retained
+    peer_store.add_indirect_peers(&clock.clock(), peer_infos[10..].iter().cloned());
+    assert_peers_in_cache(&peer_store, &peer_ids[5..], &peer_addresses[5..]);
+}
+
+/// Tests that pushing the same peers twice to the peer store does not update their
+/// place in the LruCache the second time.
+///
+/// We may learn of the same peer(s) multiple times as we exchange PeersRequest/PeersResponse
+/// with any given neighbor. However, it is not a signal of any new information about the peer,
+/// just that the peer continues to be stored by the responding neighbor.
```

Comment:
Is this really the property that we want to maintain? AFAIU this means that only peers that we don't know about can push out the old ones from the cache. IMO this cache is not about getting "fresh" information, but rather about which peers are still considered valid (so the more peers consider them valid, the better). It is not like it matters though, because it can be a gossip of garbage data anyway.

My point is that perhaps this test is not needed at all, as at least IMO it doesn't check any useful property.

Reply:
To me this test is about documenting the existing behavior so that someone does not change it unintentionally (especially with the subtlety of `peek` vs `get` on the LruCache). The comment explains my thinking in selecting the current behavior, but I agree it is not some critical property and that we may end up rethinking the eviction behavior at some point.